src/olympia/amo/models.py

```diff
@@ -94,20 +67,14 @@ def __len__(self):
         return len(list(self.__iter__()))
 
 
-class CachingRawQuerySet(RawQuerySet, caching.base.CachingRawQuerySet):
-    """A RawQuerySet with __len__ and caching."""
-
-
-# Make TransformQuerySet one of CachingQuerySet's parents so that we can do
-# transforms on objects and then get them cached.
-CachingQuerySet = caching.base.CachingQuerySet
-CachingQuerySet.__bases__ = (TransformQuerySet,) + CachingQuerySet.__bases__
-
-
-class UncachedManagerBase(models.Manager):
+class ManagerBase(models.Manager):
+    # This needs to be set so that this manager class is being used
+    # for related objects too. This helps resolving translation fields
+    # on related fields.
+    use_for_related_fields = True
```

Comment:
This also ensures we don't ignore soft-deleted items when traversing relations, if they are hidden by the `objects` manager, like we do with `Addon`.

Reply:
I'll add that to the comment.

Reply:
Added in 106b784f7e